### PR TITLE
feat(modelgen): Custom primary key support

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes

Add custom primary key support for datastore modelgen

- Fixed the bug that 'id' field is auto-generated even if the different primary key is defined
- Introduced three types of primary key and being processed as model metadata
  - `ManagedId`
  - `OptionallyManagedId`
  - `CustomId`
- Added primary key support code snippet in type declaration for Amplify JS

#### Issue #, if available

Related JS PR: https://github.com/aws-amplify/amplify-js/pull/9380

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-codegen/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] Breaking changes to existing customers are released behind a feature flag or major version update
- [ ] Changes are tested using sample applications for all relevant platforms (iOS/android/flutter/Javascript) that use the feature added/modified

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by iartemiev — CHANGES_REQUESTED** [review]
PR looks good! Just a minor change requested

File: packages/appsync-modelgen-plugin/src/visitors/appsync-javascript-visitor.ts [hunk 1]
```diff
@@ -32,7 +32,10 @@ export class AppSyncModelJavascriptVisitor<
   TRawConfig extends RawAppSyncModelJavaScriptConfig = RawAppSyncModelJavaScriptConfig,
   TPluginConfig extends ParsedAppSyncModelJavaScriptConfig = ParsedAppSyncModelJavaScriptConfig
 > extends AppSyncModelTypeScriptVisitor<TRawConfig, TPluginConfig> {
-  protected IMPORT_STATEMENTS = ['import { ModelInit, MutableModel, PersistentModelConstructor } from "@aws-amplify/datastore";'];
+  protected IMPORT_STATEMENTS = [
+    'import { ModelInit, MutableModel, PersistentModelConstructor } from "@aws-amplify/datastore";',
+    'import { ManagedIdentifier, OptionallyManagedIdentifier, CustomIdentifier } from "@aws-amplify/datastore";',
```
Comment (iartemiev) [inline packages/appsync-modelgen-plugin/src/visitors/appsync-javascript-visitor.ts:37]: For some reason I can't see my requested changes comment on GH anymore, but could these imports be combined into one? Have multiple import statements from the same module often causes linting errors.

**Review by iartemiev — APPROVED** [review]
LGTM! Thank you Aaron

**Review by david-mcafee — APPROVED** [review]
LGTM! 🎉

**Review by manueliglesias — APPROVED** [review]
LGTM

**Review by phani-srikar — APPROVED** [review]
Before releasing this, can we please start a convo with Studio that these changes are expected and get them ready. Thanks

**Comment by iartemiev** [comment]
@phani-srikar these are optional/backwards compatible changes. Studio won't need to make any changes, unless they want to start using Custom PKs

**Pull request merged**